src/main/java/com/minecolonies/blocks/BlockBarrel.java

```diff
@@ -0,0 +1,216 @@
+import com.minecolonies.items.ModItems;
+import com.minecolonies.lib.Constants;
+import com.minecolonies.util.Log;
+import net.minecraft.block.Block;
+import net.minecraft.block.material.Material;
+import net.minecraft.block.properties.PropertyInteger;
+import net.minecraft.block.state.BlockState;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.init.Items;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.BlockPos;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.common.registry.GameRegistry;
+
+import java.util.HashMap;
+import java.util.Random;
+
+public class BlockBarrel extends Block
+{
+    private static final int MIN_FULLNESS           = 0;
+    private static final int MAX_FULLNESS           = 16;
+    private static final int BARRELSTATE_FILLING    = 0;
+    private static final int BARRELSTATE_COMPOSTING = 1;
+    private static final int BARRELSTATE_DONE       = 2;
+
+    // todo: use a TileEntity to store state
+
+    private static final PropertyInteger            BARRELSTATE    = PropertyInteger.create("BARRELSTATE", BARRELSTATE_FILLING, BARRELSTATE_DONE);
+    /**
+     * The hardness this block has.
+     */
+    private static final float                      BLOCK_HARDNESS = 5F;
+    /**
+     * This blocks name.
+     */
+    private static final String                     BLOCK_NAME     = "blockBarrel";
+    /**
+     * The resistance this block has.
+     */
+    private static final float                      RESISTANCE     = 1F;
+    private static final HashMap<BlockPos, Integer> timers         = new HashMap<>();
+    private static final HashMap<BlockPos, Integer> fillings       = new HashMap<>();
+
+    public BlockBarrel()
+    {
+        super(Material.wood);
+        this.setDefaultState(this.blockState.getBaseState().withProperty(BARRELSTATE, BARRELSTATE_FILLING));
+
+        this.setTickRandomly(true);
+        initBlock();
+    }
+
+    /**
+     * initialize the block
+     */
+    private void initBlock()
+    {
+        setRegistryName(BLOCK_NAME);
+        setUnlocalizedName(String.format("%s.%s", Constants.MOD_ID.toLowerCase(), BLOCK_NAME));
+        setCreativeTab(ModCreativeTabs.MINECOLONIES);
+        GameRegistry.registerBlock(this, BLOCK_NAME);
+        setHardness(BLOCK_HARDNESS);
+        setResistance(RESISTANCE);
+    }
+
+    public void AddItemToBarrel(World worldIn, EntityPlayer playerIn, ItemStack itemStack, IBlockState state, BlockPos pos)
+    {
+        UseBarrel(worldIn, playerIn, itemStack, state, pos);
+    }
+
+    //whenever player right click to barrel call this.
+    public boolean UseBarrel(World worldIn, EntityPlayer playerIn, ItemStack itemstack, IBlockState state, BlockPos pos)
+    {
+        Log.logger.info("block activated");
+
+        int barrelState = state.getValue(BARRELSTATE);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'barrelState' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)